src/sampling.jl

```diff
@@ -109,17 +116,29 @@ end
 knuths_sample!(a::AbstractArray, x::AbstractArray; initshuffle::Bool=true) =
     knuths_sample!(Base.GLOBAL_RNG, a, x; initshuffle=initshuffle)
 
-# Fisher-Yates sampling
-#
-#   create an array of index inds = [1:n]
-#
-#   for i = 1:k
-#       swap inds[i] with a random one in inds[i:n]
-#       set x[i] = a[inds[i]]
-#   end
-#
-#   O(n) for initialization + O(k) for random shuffling
-#
+"""
+    fisher_yates_sample!([rng], a::AbstractArray, x::AbstractArray)
+
+Fisher-Yates shuffling (with early termination).
+
+Pseudo-code:
+```
+n = length(a)
+k = length(x)
+
+create an array of index inds = [1:n]
+
+for i = 1:k
+    swap inds[i] with a random one in inds[i:n]
```

Comment:
Same here. It might make more sense to put the descriptive text in comments and have to code still look like julia code.
```
# swap element `i` for another random element
inds[i] = rand(inds)

# set element `i` in `x`
x[i] = a[inds[i]]
```